common/core/desktop/include/keyman/keyboardprocessor_debug.h

```diff
@@ -48,11 +63,12 @@ typedef struct {
  *
  * Used in all event types except KM_KBP_DEBUG_BEGIN, KM_KBP_DEBUG_END.
  */
+
```

Comment:
So, by "event types", I assume that "all event types [except...]" means that near-all debugging event types are specific to individual rule applications?

I'm drawing that inference from the `store_offsets` field and its limited size, which is stated to be specific to evaluation of individual rules in the earlier comment.  I'm pretty certain this must be the case (at least, for events that need a `kmx_info` item), as otherwise the two comments conflict with each other.

Reply:
Yes, that's correct.